tests/tests/connectors/test-class-connector-posts.php

```diff
@@ -2,11 +2,127 @@
 namespace WP_Stream;
 
 class Test_WP_Stream_Connector_Posts extends WP_StreamTestCase {
+	/**
+	 * Holds local timestamp in "Y-m-d H:i:s" format
+	 *
+	 * @var string
+	 */
+	private $date;
 
-	public function test_reminder() {
-		$this->markTestIncomplete(
-			'Not implemented yet.'
+	/**
+	 * Holds GMT timestamp in "Y-m-d H:i:s" format
+	 *
+	 * @var string
+	 */
+	private $date_gmt;
+
+	/**
+	 * Runs before each test.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->plugin->connectors->unload_connectors();
+
+		// Set static timestamps.
+		$this->date     = '2007-07-04 12:30:00';
+		$this->date_gmt = get_gmt_from_date( $this->date );
+
+		// Make partial of Connector_Posts class, with mocked "log" function.
+		$this->mock = $this->getMockBuilder( Connector_Posts::class )
+			->setMethods( [ 'log' ] )
+			->getMock();
+
+		// Register connector.
+		$this->mock->register();
+	}
+
+	/**
+	 * Tests "transition_post_status" callback function.
+	 */
+	public function test_callback_transition_post_status() {
+		// Set expected calls for the Mock.
+		$this->mock->expects( $this->once() )
+			->method( 'log' )
+			->with(
+				$this->equalTo(
+					_x(
+						'"%1$s" %2$s updated',
+						'1: Post title, 2: Post type singular name',
+						'stream'
+					)
+				),
+				$this->equalTo(
+					array(
+						'post_title'    => 'Test post',
+						'singular_name' => 'post',
+						'post_date'     => $this->date,
+						'post_date_gmt' => $this->date_gmt,
+						'new_status'    => 'publish',
+						'old_status'    => 'new',
+						'revision_id'   => null,
+					)
+				),
+				$this->greaterThan( 0 ),
+				$this->equalTo( 'post' ),
+				$this->equalTo( 'updated' )
```

Comment:
@dero `updated` is used as the context for a lot of log calls. I agree that this should be changed, however saddle up because you'll be running in to mismatch context issues a lot.

Reply:
> because you'll be running in to mismatch context issues

@kidunot89 Can you please expand on what you mean by this?

Reply:
Not too much to expand on just that `updated` is used a lot as the ~context~ _action_ and it doesn't always express the what's happening.